src/server/NetworkingServer.cpp

```diff
@@ -168,10 +175,11 @@ void NetworkingServer::handleReliableCommand(sf::Socket::Status status, sf::Pack
 }
 
 void NetworkingServer::deleteClient(sf::TcpSocket* client, std::list<sf::TcpSocket*> clients) {
-    int client_id = mClientSocketsToIds.get(client);
+    int deleted_client_id = mClientSocketsToIds.get(client);
     clients.remove(client);
     mClientSocketsToIds.erase(client);
-    mRemovedClientIds.add(client_id);
+    EventController::getInstance().queueEvent(
```

Comment:
I dont really understand shared_ptr's but we use em everywhere

Reply:
It keeps track of how many copies of the pointer exist and deletes it when there are none left. This avoids "memory leaks and dangling Pointers" https://thispointer.com/learning-shared_ptr-part-1-usage-details/

Reply:
is there any downside?